x12/transactions/x12_271_005010X279A1/parsing.py

```diff
@@ -0,0 +1,425 @@
+    SUBSCRIBER_ADDITIONAL_ELIGIBILITY = "loop_2115c"
+    SUBSCRIBER_RELATED_BENEFIT_NAME = "loop_2120c"
+    DEPENDENT = "loop_2000d"
+    DEPENDENT_NAME = "loop_2100d"
+    DEPENDENT_ELIGIBILITY = "loop_2110d"
+    DEPENDENT_ADDITIONAL_ELIGIBILITY = "loop_2115d"
+    DEPENDENT_RELATED_BENEFIT_NAME = "loop_2120d"
+    FOOTER = "footer"
+
+
+SUBSCRIBER_LOOPS: Set[TransactionLoops] = {
+    TransactionLoops.SUBSCRIBER,
+    TransactionLoops.SUBSCRIBER_NAME,
+    TransactionLoops.SUBSCRIBER_ELIGIBILITY,
+    TransactionLoops.SUBSCRIBER_ADDITIONAL_ELIGIBILITY,
+    TransactionLoops.SUBSCRIBER_RELATED_BENEFIT_NAME,
+}
+
+DEPENDENT_LOOPS: Set[TransactionLoops] = {
+    TransactionLoops.DEPENDENT,
+    TransactionLoops.DEPENDENT_NAME,
+    TransactionLoops.DEPENDENT_ELIGIBILITY,
+    TransactionLoops.DEPENDENT_ADDITIONAL_ELIGIBILITY,
+    TransactionLoops.DEPENDENT_RELATED_BENEFIT_NAME,
+}
+
+
+def _get_info_source(context) -> Dict:
+    """Returns the current information source"""
+
+    return context.transaction_data[TransactionLoops.INFORMATION_SOURCE][-1]
+
+
+def _get_info_receiver(context) -> Dict:
+    """Returns the current information receiver"""
+
+    info_source = _get_info_source(context)
+    return info_source[TransactionLoops.INFORMATION_RECEIVER][-1]
+
+
+def _get_subscriber(context) -> Dict:
+    """Returns the current subscriber"""
+    info_receiver = _get_info_receiver(context)
+    return info_receiver[TransactionLoops.SUBSCRIBER][-1]
+
+
+def _get_dependent(context) -> Dict:
+    """Returns the current dependent"""
+    subscriber = _get_subscriber(context)
+    return subscriber[TransactionLoops.DEPENDENT][-1]
+
+
+def _get_subscriber_eligibility(context) -> Dict:
+    """Returns the current subscriber eligibility"""
+    subscriber = _get_subscriber(context)
+    return subscriber[TransactionLoops.SUBSCRIBER_NAME][
+        TransactionLoops.SUBSCRIBER_ELIGIBILITY
+    ][-1]
+
+
+def _get_dependent_eligibility(context) -> Dict:
+    """Returns the current dependent eligibility"""
+    dependent = _get_dependent(context)
+    return dependent[TransactionLoops.DEPENDENT_NAME][
+        TransactionLoops.DEPENDENT_ELIGIBILITY
+    ][-1]
+
+
+@match("ST")
+def set_header_loop(context: X12ParserContext) -> None:
+    """
+    Sets the transaction set header loop.
+    Creates initial data structures for the record
+
+    :param context: The X12Parsing context which contains the current loop and transaction record.
+    """
+    context.transaction_data[TransactionLoops.HEADER] = {"bht_segment": None}
+
+    context.set_loop_context(
+        TransactionLoops.HEADER, context.transaction_data[TransactionLoops.HEADER]
+    )
+
+
+@match("HL", {"hierarchical_level_code": "20"})
+def set_info_source_loop(context: X12ParserContext) -> None:
+    """
+    Sets the information source loop (Loop 2000A).
+
+    :param context: The X12Parsing context which contains the current loop and transaction record.
+    """
+    info_source = {
```

Comment:
Here's an example of defining the fields in a loop.
The `[]` are required due to the code we have in the ModelReader.  For the moment this is the easiest way to inform the reader that the model is a list field. In a future PR we could revisit parsing the Pydantic metadata. 

The issue with the Pydantic metadata approach is that type information is conveyed as a "string" rather than a concrete type we can evaluate.